vignettes/ABN_inference.Rmd

```diff
@@ -0,0 +1,215 @@
+---
+title: "Inference with ABN"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Inference with ABN}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+```
+
+This vignette provides an overview of performing inference with ABN.
+
+## Background
+
+In Additive Bayesian Networks, inference refers to the process of computating the expectation of a subset of variables, given observed or known values of other variables in the network.
+
+**Example**: let's consider the following example. 
+
+```{r, message=FALSE}
+library(abn)
+
+# Load an illustrative example
+mydat <- ex1.dag.data
+
+# Specify the distributions of the nodes
+mydists <- list(b1="binomial", 
+                p1="poisson",
+                g1="gaussian", 
+                b2="binomial", 
+                p2="poisson",
+                b3="binomial", 
+                g2="gaussian", 
+                b4="binomial", 
+                b5="binomial", 
+                g3="gaussian") 
+
+# Run abn
+max.par <- 4 # set the same max parents for all nodes
+mycache <- buildScoreCache(data.df = mydat, 
+                           data.dists = mydists,
+                           method = "bayes",max.parents = max.par) 
+mp.dag <- mostProbable(score.cache = mycache)
+myfit <- fitAbn(object = mp.dag)
+plot(myfit)
+```
+
+We would like to compute the conditional expectation $\mathbb{E}(g2|b2="yes",b5="yes")$.
+
+Some useful definitions:
+
+- **Hypothesis**: the hypothesis refers to the set of variables whose values we want to infer ($g2$ in the example above).
+
+- **Evidence**: evidence refers to the set of known or observed values of other variables in the network ($b2$ and $b5$ in the example above). 
+
+## Methodology
+
+The inference procedure we propose is an iterative process, which main goal consists in refining the estimate of the hypothesis based on both its parents and children:
+
+1. Estimate the hypothesis using the parents, $\mathbb{E}(g2|b2 = "yes")$ in our example,
+
+2. Update the estimate by considering the children, $\mathbb{E}(g2|b5="yes")$ in our example is updated based on the initial estimate (prior) from Step 1,
+
+3. Update the estimate by averaging for a more balanced estimate, $\mathbb{E}(g2|b2 = "yes",b5="yes") = (1/2)\times \mathbb{E}(g2|b2 = "yes") + (1/2) \times \mathbb{E}(g2|b5 = "yes").$
+
+4. Iterate until convergence: the updated estimate from Step 3 becomes the priors for the next step (Step 2). 
+
```

Comment:
Are there any assumptions? For example, parent-child nodes should not have a time-dependent relationship.